setup.py

```diff
@@ -23,7 +23,7 @@
         # Use prompt_toolkit 3.0.18, because of the `in_thread` option.
         "prompt_toolkit>=3.0.18,<3.1.0",
         "pygments",
-        "black",
+        "black;sys_platform == 'win32' and platform_python_implementation != 'PyPy'",
```

Comment:
How does this actually get interpreted? Does it mean that "black will only be installed if the platform is Windows and the interpreter is not Pypy?" If that is true, does it mean that it won't be installed on Linux at all?

Reply:
I took the page from Christoph Gohlke. It is to solve the problem on Windows + PyPy, as the "Black but in fact Pywin32" issue is on Windows + PyPy only

Reply:
I just tried this on Linux. This particular line makes it so that black doesn't get installed, which is not desirable. What if we only exclude pypy regardless of the platform?

Reply:
With the above change, I'm also good if we move it into an ``extras_requires`` block, named e.g., ``"all"``.

Reply:
surely "extras_requires" was making more sense to me. let's so do this

Reply:
like this  ?

```
    extras_require={
        "ptipython": ["ipython"],  # For ptipython, we need to have IPython
        "all": ["black"],
    },
```